src/main/groovy/org/jenkins/ci/plugins/buildtimeblame/analysis/MapFixStackedAreaRenderer.groovy

```diff
@@ -0,0 +1,46 @@
+package org.jenkins.ci.plugins.buildtimeblame.analysis
+
+import org.jfree.chart.entity.EntityCollection
+import org.jfree.chart.renderer.category.StackedAreaRenderer
+import org.jfree.data.category.CategoryDataset
+
+import java.awt.Shape
+import java.awt.geom.GeneralPath
+import java.awt.geom.PathIterator
+
+class MapFixStackedAreaRenderer extends StackedAreaRenderer {
+    /**
+     * Breaks GeneralPath hotspots from StackedAreaRenderer into separate paths for
+     * each polygon. This workaround lets JFreeChart build the correct image map.
+     */
+    @Override
+    protected void addItemEntity(EntityCollection entities,
+            CategoryDataset dataset, int row, int column, Shape hotspot) {
+        PathIterator pi = hotspot.getPathIterator(null, 1.0)
+        float[] coords = new float[6]
+        GeneralPath subPath = new GeneralPath(pi.getWindingRule())
+
+        while (!pi.isDone()) {
+            int type = pi.currentSegment(coords)
+
+            switch (type) {
+                case PathIterator.SEG_MOVETO:
+                    subPath.moveTo(coords[0], coords[1])
+                    break
+                case PathIterator.SEG_LINETO:
+                    subPath.lineTo(coords[0], coords[1])
+                    break
```

Comment:
How hard would it be to test these two cases? 

To be honest, I don't follow too much of how this works but, the results look good.

Reply:
These cases are indirectly covered by the 'should split compound paths' test. The `pathsMatch` argument constraint checks that moveTo/lineTo are getting called correctly while building the subPath eventually passed to `entityCollection.add`.

I could add another test with GeneralPath mocked, and interactions that directly confirm these cases are handled correctly.

Reply:
Oh, I see. Checking the results like your existing test is better than mocking. It just takes some IDE navigating to follow it.